tests/unit/remoteworker/test_remote_worker.py

```diff
@@ -0,0 +1,116 @@
+import mock
+
+from unittest import TestCase
+
+from scripts.workers.remote_submission_worker import (
+    make_request,
+    get_message_from_sqs_queue,
+    delete_message_from_sqs_queue,
+    get_submission_by_pk,
+    get_challenge_phases_by_challenge_pk,
+    get_challenge_by_queue_name,
+    get_challenge_phase_by_pk,
+    update_submission_data,
+    update_submission_status,
+    return_url_per_environment,
+)
+
+
+class BaseTestClass(TestCase):
+    def setUp(self):
+        self.url = "/test/url"
+        self.submission_pk = 1
+        self.challenge_pk = 1
+        self.challenge_phase_pk = 1
+        self.data = {"test": "data"}
+        self.headers = {"Authorization": "Token test_token"}
+
+
+@mock.patch("scripts.workers.remote_submission_worker.AUTH_TOKEN", "test_token")
+@mock.patch("scripts.workers.remote_submission_worker.requests")
+class MakeRequestTestClass(BaseTestClass):
+
+    def test_make_request_get(self, mock_make_request):
+        make_request(self.url, "GET")
+        mock_make_request.get.assert_called_with(url=self.url, headers=self.headers)
+
+    def test_make_request_put(self, mock_make_request):
+        make_request(self.url, "PUT", data=self.data)
+        mock_make_request.put.assert_called_with(url=self.url, headers=self.headers, data=self.data)
+
+    def test_make_request_patch(self, mock_make_request):
+        make_request(self.url, "PATCH", data=self.data)
+        mock_make_request.patch.assert_called_with(url=self.url, headers=self.headers, data=self.data)
+
+
+@mock.patch("scripts.workers.remote_submission_worker.QUEUE_NAME", "evalai_submission_queue")
+@mock.patch("scripts.workers.remote_submission_worker.return_url_per_environment")
+@mock.patch("scripts.workers.remote_submission_worker.make_request")
+class APICallsTestClass(BaseTestClass):
+
+    def test_get_message_from_sqs_queue(self, mock_make_request, mock_url):
+        self.url = "/api/jobs/challenge/queues/evalai_submission_queue/"
```

Comment:
@KhalidRmb why do we still have `urls` hardcoded? can't we just define `get` methods in base class for fetching the required url?

Reply:
I'm not sure what you mean. I'm hard coding them because a different url is used for each test case.

Reply:
So what I want is `get` methods in base class to fetch the required URL, hardcoding them in each test method is not a good practice. Methods in base class can be reused for other methods. Did you get my point?

Reply:
@RishabhJain2018 your thoughts!